contributing/guide-observability.md

```diff
@@ -120,6 +120,56 @@ logs don't show up in production normally.
 For the question of what constitutes an error, errors should be actionable by a
 human. See the [Dave Cheney article] on this topic for some more discussion.
 
+## Good Errors
+
+An error message is good if it communicates
+
+1. what went wrong
+2. where it went wrong
+3. for which inputs
+
+and if it is concise without redundancies and at best displayable in one line.
+
+The general goal of an error message is to be actionable for the user who is trying
+to avoid the error in the future.
+
+Examples for bad error messages:
+- `cannot resolve package dependencies: Invalid Semantic Version`
+- `cannot parse file foo.yaml: failed to parse YAML of file foo.yaml`
+
+Examples for how to improve them:
+- `cannot resolve package dependencies in crossplane.yaml: invalid Semantic Version "main" on package "provider-kubernetes"`
+- `cannot parse file: failed to parse YAML of file foo.yaml` or the inverse.
+
+General rules:
+1. follow the pattern `generic error cause: more details: even more details`.
```

Comment:
maybe more specific: add error context information that is bound it its current layer of abstraction. Taking your example of above: `error parsing package depencies content: failed to parse YAML of file foo.yaml`

Reply:
here:
- high level layer: `error parsing package depencies content` happens at the layer that is trying to deserialize into concrete objects (package dependency objects)
- low level layer: `failed to parse YAML of file foo.yaml`: happens at the layer that deals with opening files and parsing generic yaml

Reply:
added